api/types.proto

```diff
@@ -713,3 +713,34 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	// CONTAINER secrets are user-specified secrets that should be injected
+	// into the container as an environment variable or file.
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
```

Comment:
After some IRL discussion with @diogomonica, we can go with 2 options:

1.  We can have a `NodeInternalSecret` type for things like networks, and registry auth.  Since secrets can have restrictions on them, we can also restrict which nodes have access to which secrets.  Since we're enforcing that user secrets have to be ascii/digits, in order to avoid name conflicts with user-specified secrets, we can require that internal secrets start with 🐳 or something (if the annotations can accept unicode).  And then secrets work the same way for everything.

1.  Another option is to have different concrete types.  So maybe `Secret` can have one of `[]*ContainerSecretValue` or one of `[]*NodeInternalSecretValue`, which has better type safety and would allow us to have different specs or definitions of these different types of secret.  But it would mean a larger number of defined objects and types.

/cc @diogomonica @aaronlehmann @stevvooe